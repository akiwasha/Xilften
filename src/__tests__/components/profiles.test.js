import React from 'react';
import { render } from '@testing-library/react';
import { Profiles } from '../../components';

describe('<Profiles />', () => {
  it('renders the <Profiles /> with the populated data', () => {
    const { container, getByText, getByTestId } = render(
      <Profiles>
        <Profiles.Title>Qui regarde?</Profiles.Title>
        <Profiles.List>
          <Profiles.User onClick={() => {}}>
            <Profiles.Picture
              src="/images/plv.png"
              data-testid="profile-picture"
            />
            <Profiles.Name>Pierre-Laurent Vincent</Profiles.Name>
          </Profiles.User>
        </Profiles.List>
      </Profiles>
    );
    expect(getByText('Qui regarde?')).toBeTruthy();
    expect(getByTestId('profile-picture')).toBeTruthy();
    expect(getByText('Pierre-Laurent Vincent')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the <Profiles /> with the populated data', () => {
    const { container, getByText, getByTestId } = render(
      <Profiles>
        <Profiles.Title>Qui regarde?</Profiles.Title>
        <Profiles.List>
          <Profiles.User onClick={() => {}}>
            <Profiles.Picture
              src="/images/plv.png"
              data-testid="profile-picture-misc"
            />
            <Profiles.Name>Pierre-Laurent Vincent</Profiles.Name>
          </Profiles.User>
        </Profiles.List>
      </Profiles>
    );
    expect(getByText('Qui regarde?')).toBeTruthy();
    expect(getByTestId('profile-picture-misc')).toBeTruthy();
    expect(getByText('Pierre-Laurent Vincent')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
