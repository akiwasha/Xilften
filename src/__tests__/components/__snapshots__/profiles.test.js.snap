// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profiles /> renders the <Profiles /> with the populated data 1`] = `
<div
  class="profiles__Container-rdzfz4-0 ctjKKt"
>
  <h1
    class="profiles__Title-rdzfz4-1 dYeiMT"
  >
    Qui regarde?
  </h1>
  <ul
    class="profiles__List-rdzfz4-2 iweZju"
  >
    <li
      class="profiles__Item-rdzfz4-5 krfgQo"
    >
      <img
        class="profiles__Picture-rdzfz4-4 hyBtrx"
        data-testid="profile-picture"
        src="/images/users//images/plv.png.png"
      />
      <p
        class="profiles__Name-rdzfz4-3 bqOkZC"
      >
        Pierre-Laurent Vincent
      </p>
    </li>
  </ul>
</div>
`;

exports[`<Profiles /> renders the <Profiles /> with the populated data 2`] = `
<div
  class="profiles__Container-rdzfz4-0 ctjKKt"
>
  <h1
    class="profiles__Title-rdzfz4-1 dYeiMT"
  >
    Qui regarde?
  </h1>
  <ul
    class="profiles__List-rdzfz4-2 iweZju"
  >
    <li
      class="profiles__Item-rdzfz4-5 krfgQo"
    >
      <img
        class="profiles__Picture-rdzfz4-4 hyBtrx"
        data-testid="profile-picture-misc"
        src="/images/users//images/plv.png.png"
      />
      <p
        class="profiles__Name-rdzfz4-3 bqOkZC"
      >
        Pierre-Laurent Vincent
      </p>
    </li>
  </ul>
</div>
`;
